client/blocks/login/two-factor-authentication/verification-code-input.jsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLink from 'components/external-link';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
+import FormButton from 'components/forms/form-button';
+import FormTextInput from 'components/forms/form-text-input';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormInputValidation from 'components/forms/form-input-validation';
+import Card from 'components/card';
+import { localize } from 'i18n-calypso';
+import { loginUserWithTwoFactorVerificationCode } from 'state/login/actions';
+import {
+	getTwoFactorAuthId,
+	getTwoFactorAuthNonce,
+} from 'state/login/selectors';
+
+class VerificationCodeInput extends Component {
+	static propTypes = {
+		rememberMe: PropTypes.bool.isRequired,
+		twoStepId: PropTypes.string.isRequired,
+		twoStepNonce: PropTypes.string.isRequired,
+	};
+
+	state = {
+		twoStepCode: '',
+		error: null
+	};
+
+	onChangeField = ( event ) => {
+		if ( event.target.type === 'checkbox' ) {
+			this.setState( {
+				[ event.target.name ]: event.target.checked
+			} );
+			return;
+		}
+		// Reset the error state if the user updates the field after an error coming
+		// from the state
+		this.setState( {
+			[ event.target.name ]: event.target.value,
+			error: null
+		} );
+	};
+
+	onCodeSubmit = ( event ) => {
+		const { twoStepId, twoStepNonce, rememberMe } = this.props;
+		const { twoStepCode } = this.state;
+		event.preventDefault();
+		this.props.loginUserWithTwoFactorVerificationCode( twoStepId, twoStepCode, twoStepNonce, rememberMe ).then( () => {
+			this.props.onSuccess( this.state );
+		} );
+	};
+
+	render() {
+		const { translate } = this.props;
+		const isError = !! this.state.error;
+		let errorText = this.state.error;
+		if ( isError ) {
+			errorText = translate( 'Invalid verification code' );
+		}
```

Comment:
I don't see a reason to assign `errorText` to `this.state.error` at any point if we always use the same message. What do you think about using the following?

```js
const isError = !! this.state.error;
const errorText = isError && translate( 'Invalid verification code' );
```